### ISSUE TITLE
utils.ainput doesn't work in jupyter since commit e0cee3923a8809e3d182f3d7651d3ba51641a38b

### ISSUE THREAD

**Issue by ivanlonel** [Gobot1234/steam.py#295]
### Summary

Replacing builtin input() for asyncio.StreamReader() broke interactivity in notebooks.

### Reproduction Steps

In a Kaggle notebook running Python 3.10, I run this to install steam.py:
```python
%pip install -U "steamio @ git+https://github.com/Gobot1234/steam.py@main"
```

Then I run this code, replacing my user and password:
```python
import steam

class MyClient(steam.Client):
    async def on_ready(self) -> None:
        print(f"We have logged in as {self.user}")

client = MyClient()

await client.login("<my username>", "<my password>")
```

### Expected Results

Actually let me type in the code and continue execution flow.
```
Please enter a Steam guard code
>>>  *****
We have logged in as <user>
```

### Actual Results

After printing the prompt for the Steam guard code, it immediately raised an Exception before I had the chance to type in the code, then it hung up forever as if it was still waiting for an input.
```
Please enter a Steam guard code
>>> 
Exception in thread Thread-5 (_read_line_from_stdin):
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/opt/conda/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/conda/lib/python3.10/site-packages/steam/utils.py", line 274, in _read_line_from_stdin
    raise EOFError
EOFError
```

### System Information

```
python version: 3.10.12
steam.py version: 1.0.0a2876+gbb5463b
aiohttp version: 3.8.4
betterproto version: 2.0.0b6
operating system info: Linux-5.15.120+-x86_64-with-glibc2.31
```

### Checklist

- [X] I have searched the issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my username, password, shared/identity secrets and refresh/access tokens from display, if visible.

**Comment by Gobot1234** [comment]
jupyter appears to be patching input to call something else `<bound method Kernel.raw_input of <ipykernel.ipkernel.IPythonKernel object at 0x7f14f2bac820>>` not entirely sure what can be done here. Reverting back to just calling input using to_thread would be a regression for everyone else.

**Comment by Gobot1234** [comment]
If you want to fix this yourself you can just override the `code` method on `Client` like so
```py
class Client(steam.Client):
    async def code(self) -> str:
        return await asyncio.to_thread(input, "Please enter a Steam Guard Code\n>>> ")
```

**Comment by ivanlonel** [comment]
Thanks a lot for you quick replies, @Gobot1234.

I copied the `code` method you provided to my Client subclass and the prompt does work now, but it still hangs up after I enter the code, without executing the `print` inside `on_ready`. I also tested it in an actual script running locally on Python 3.11, and the behavior is the same (so no longer a Jupyter problem), whether or not I override the `code` method.

One thing I noticed is the push notification I get from my Steam app now is `New sign in request for <user>`, and when I tap it, it asks me to just confirm if I'm actually the one trying to login (doesn't matter if I confirm it before or after entering the code in the notebook, it always hangs). With version 0.9.9, the push says `Login to Steam with code *****`, and the app doesn't ask me to confirm anything, it only shows the code.

Regarding the Jupyter prompt issue specifically, overriding `code` in the client is a good enough solution for me. Specially because running steam.py on a notebook is probably a very niche use case.
